backend/globaleaks/handlers/admin/user.py

```diff
@@ -143,7 +197,8 @@ def db_admin_update_user(session, tid, user_session, user_id, request, language)
 
     # The various options related in manage PGP keys are used here.
     parse_pgp_options(user, request)
-
+    if user.role != request['role'] and tid == 1:
+        generate_analyst_key_pair(session, user, user_session)
```

Comment:
As this code need to be executed only during wizard or at migration time, this should be moved. It will in fact never needed to generate a new key when creating a new user but eventually the existing key generated on the system will need to be assigned to the new user.